bluer/src/mesh/types.rs

```diff
@@ -0,0 +1,98 @@
+pub use btmesh_common::{
+    address::{Address, UnicastAddress},
+    InsufficientBuffer, ModelIdentifier,
+};
+use btmesh_common::{opcode::Opcode, ParseError};
+use btmesh_models::{Message as ConcreteMessage, Model as ConcreteModel};
+use core::fmt::Debug;
+
+/// Bluetooth Mesh Message
+pub trait Message {
+    /// Returns opcode of the message
+    fn opcode(&self) -> Opcode;
+    /// Emit message parameters
+    fn emit_parameters(&self, xmit: &mut Vec<u8>);
+}
+
+/// Bluetooth Mesh Model
+pub trait Model: Sync + Send + Debug {
+    /// Returns model identifier
+    fn identifier(&self) -> ModelIdentifier;
+    /// Returns whether model supports subscription
+    fn supports_subscription(&self) -> bool;
```

Comment:
You could provide a default implementation that returns false.
Results in less boilerplate code for your users.